src/app/modules/chip-filter/chip-filter.component.ts

```diff
@@ -0,0 +1,338 @@
+import {
+  Component,
+  HostListener,
+  forwardRef,
+  Input,
+  Output,
+  EventEmitter,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  ViewEncapsulation
+} from '@angular/core';
+import {NG_VALUE_ACCESSOR, ControlValueAccessor} from '@angular/forms';
+import {ListItem, IDropdownSettings} from './chip-filter.model';
+import {ChipFilterPipe} from './chip-filter.pipe';
+import {faTimes} from '@fortawesome/free-solid-svg-icons';
+
+export const DROPDOWN_CONTROL_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => AppChipFilterComponent),
+  multi: true
+};
+
+const noop = () => {
+};
+
+@Component({
+  selector: 'app-chip-filter',
+  templateUrl: './chip-filter.component.html',
+  styleUrls: ['./chip-filter.component.scss'],
+  providers: [DROPDOWN_CONTROL_VALUE_ACCESSOR],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None
+})
+export class AppChipFilterComponent implements ControlValueAccessor {
+  public _settings: IDropdownSettings;
+  public _data: Array<ListItem> = [];
+  private _sourceDataType = null;
+  private _sourceDataFields: Array<String> = [];
+  public selectedItems: Array<ListItem> = [];
+  public icon = faTimes;
+  public filter: ListItem = new ListItem(this.data);
+  public defaultSettings: IDropdownSettings = {
+    singleSelection: false,
+    idField: 'id',
+    textField: 'text',
+    disabledField: 'isDisabled',
+    enableCheckAll: true,
+    selectAllText: 'Select All Providers',
+    unSelectAllText: 'UnSelect All Providers',
+    allowSearchFilter: false,
+    limitSelection: -1,
+    clearSearchFilter: true,
+    maxHeight: 197,
+    itemsShowLimit: 999999999999,
+    searchPlaceholderText: 'Search',
+    noDataAvailablePlaceholderText: 'No data available',
+    closeDropDownOnSelection: false,
+    showSelectedItemsAtTop: false,
+    defaultOpen: false,
+    allowRemoteDataSearch: false
+  };
+
+  @Input() hintMessage: string;
```

Comment:
Nice work. looks really nice. I remember working with Josh and we want to combine Inputs as one object/model. Kind of how it is for the Calendar component.

Reply:
That way we can just pass one Model Object instead of many inputs.

Reply:
Good feedback I'll check it out!

Reply:
@JohnYuBellese Just moved the label and error message config stuff into the settings configuration model. I left out disabled so you don't have to mutate the config object to disable from the consuming component.